shimmingtoolbox/dicom_to_nifti.py

```diff
@@ -54,6 +55,44 @@ def dicom_to_nifti(path_dicom, path_nifti, subject_id='sub-01', path_config_dcm2
     subprocess.run(['dcm2bids', '-d', path_dicom, '-o', path_nifti, '-p', subject_id, '-c', path_config_dcm2bids],
                    check=True)
 
+    # In the special case where a phasediff should be created but the filename is phase instead. Find the file and
+    # rename it
+    # Go in the fieldmap folder
+    path_fmap = os.path.join(path_nifti, subject_id, 'fmap')
+    if os.path.exists(path_fmap):
+        # Make a list of the json files in fmap folder
+        file_list = []
+        [file_list.append(os.path.join(path_fmap, f)) for f in os.listdir(path_fmap)
+         if os.path.splitext(f)[1] == '.json']
+        file_list = sorted(file_list)
+
+        for fname_json in file_list:
+            is_renaming = False
+            # Open the json file
+            with open(fname_json) as json_file:
+                json_data = json.load(json_file)
+                # Make sure it is a phase data and that the keys EchoTime1 and EchoTime2 are defined and that
+                # sequenceName's last digit is 2 (refers to number of echoes when using dcm2bids)
+                if ('ImageType' in json_data) and ('P' in json_data['ImageType']) and \
+                   ('EchoTime1' in json_data) and ('EchoTime2' in json_data) and \
+                   ('SequenceName' in json_data) and (int(json_data['SequenceName'][-1]) == 2):
+                    # Make sure it is not already named phasediff
+                    if len(os.path.basename(fname_json).split(subject_id, 1)[-1].rsplit('phasediff', 1)) == 1:
+                        # Split the filename in 2 and remove phase
+                        file_parts = fname_json.rsplit('phase', 1)
+                        if len(file_parts) == 2:
+                            # Stitch the filename back together making sure to remove any digits that could be after
+                            # 'phase'
+                            digits = '0123456789'
+                            fname_new_json = file_parts[0] + 'phasediff' + file_parts[1].lstrip(digits)
+                            is_renaming = True
+            # Rename the json file
+            if is_renaming:
+                os.rename(fname_json, fname_new_json)
+                fname_nifti_new = os.path.splitext(fname_new_json)[0] + '.nii.gz'
```

Comment:
what if the input file is a nii instead of nii.gz?

Reply:
I can account for it, however I think dcm2bids always outputs the '.nii.gz' version. So I'm not sure how I could test it. It will be there if dcm2bids ever puts the option to output a .nii

Reply:
i see, well in that case let's leave it for now
but i know many people who work with nii, so in case they use ST bypassing dcm2bids, then it will fail

Reply:
If they do bypass dicom_to_nifti, then the renaming part of the code won't be executed as well.

Reply:
ok in that case let's not worry too much with it then-- i see you did add some code supporting nii, but your arguments are solid enough to not add these lines of code (it might create confusion from someone external reading the code-- moreover, if we want fulll NII compatibility maybe more code needs to be changed)